lib/dune

```diff
@@ -23,7 +23,6 @@
    write.mli
    read.mli
    safe.mli
-   pretty.mli
```

Comment:
Any reason why this is removed?

Reply:
This is very much a deliberate choice. That file is never actually seen by the compiler since it is not included in `yojson.cppo.mli` since the compiler only ever sees `write2.mli` and `pretty.mli` is not part of the public interface.

To wit, check out the commit right before and notice the code compiling just fine despite `pretty.mli` referencing a `json` type that was removed.

I think the whole CPPO magic is extremely intransparent and should be simplified so I opted to at least remove parts that don't seem to have any function to decrease confusion.

Reply:
Ah my bad! This is unfortunate indeed, let's remove it!